### PR TITLE
Build: updates

### PR DESCRIPTION
**Pull request by PatrykMis**
* Update Java base to 17
* update Kotlin to 1.8.22
* update Bouncycastle
* update all dependencies
* follow the manifest on jni libs packaging
* enhance app optimization by specifying resource configurations, excluding dependencies info and unnessesary files
* Remove obsolete SDK check as we are already using minSdk 26

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -17,11 +17,17 @@ android {
...
+            "en",
+        ))
 
         vectorDrawables.useSupportLibrary = true
     }
 
     buildTypes {
+        debug {
+            applicationIdSuffix = ".dev"
+        }
```
Comment (Axelen123) [inline app/build.gradle.kts:30]: Why was this added?

Comment (PatrykMis) [reply app/build.gradle.kts:30]: Because it will be easier to test new features while having installed stable, "release" version for everyday use. But if you wish I can delete it and rebase.

Comment (Axelen123) [reply app/build.gradle.kts:30]: I don't think we need it. Work profiles and emulators are a thing

File: app/build.gradle.kts [hunk 2]
```diff
@@ -17,11 +17,17 @@ android {
         targetSdk = 33
         versionCode = 1
         versionName = "0.0.1"
+        resourceConfigurations.addAll(listOf(
+            "en",
+        ))
```
[the remark this answers is not in the document]
Comment (PatrykMis) [reply app/build.gradle.kts:22]: <https://developer.android.com/build/shrink-code#unused-alt-resources>

[the remark this answers is not in the document]
Comment (Axelen123) [reply app/build.gradle.kts:22]: Alright. Makes sense

File: app/build.gradle.kts (hunk 1 above)
Comment (PatrykMis) [reply app/build.gradle.kts]: Ok, so rebased without it.

**Review by Axelen123 — APPROVED** [review]
